arch/x86/include/asm/kfence.h

```diff
@@ -0,0 +1,173 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+#ifndef _ASM_X86_KFENCE_H
```

Comment:
Is "asm" a good place for this header?

Reply:
That's where all arch-specific code goes. Where else should it go?